### PR TITLE
remove deprecated features

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by xabbuh** [comment]
need to investigate the test failures

Status: Needs work

File: src/Symfony/Component/Form/Tests/Extension/Validator/EventListener/ValidationListenerTest.php [hunk 1]
```diff
@@ -64,7 +65,7 @@ protected function setUp()
         $this->params = array('foo' => 'bar');
     }
 
-    private function getConstraintViolation($code = null, $constraint = null)
+    private function getConstraintViolation($code = null, Constraint $constraint)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Tests/Extension/Validator/EventListener/ValidationListenerTest.php]: `= null` should removed on $code isn't it?

**Comment by nicolas-grekas** [comment]
@webmozart if you could help on one thing for Symfony 3.0 it could be this one: removing the remaining deprecated code from 3.0. I had a hard time trying to do so, now @xabbuh is, and still we're not at it. Would you like to help on this?

**Comment by nicolas-grekas** [comment]
Rebase needed here

File: src/Symfony/Component/Form/Tests/Extension/Validator/EventListener/ValidationListenerTest.php [hunk 2]
```diff
@@ -16,6 +16,7 @@
 use Symfony\Component\Form\Extension\Validator\Constraints\Form;
 use Symfony\Component\Form\Extension\Validator\EventListener\ValidationListener;
 use Symfony\Component\PropertyAccess\PropertyPath;
+use Symfony\Component\Validator\Constraint;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Tests/Extension/Validator/EventListener/ValidationListenerTest.php]: To be removed

**Comment by nicolas-grekas** [comment]
@xabbuh will you be able to finish this  PR is the coming days? Don't hesitate to call for help if needed.

**Comment by xabbuh** [comment]
@nicolas-grekas I think so. I will have to look a bit deeper into why the Form tests are failing, but I hope to be able to do that tonight.

**Comment by xabbuh** [comment]
I am closing here as I didn't manage to make the test pass.

**Pull request closed without merging**